pylint/plugins/hass_enforce_type_hints.py

```diff
@@ -35,371 +34,330 @@ class TypeHintMatch:
     "x_of_y_of_z_comma_a": re.compile(r"^(\w+)\[(\w+)\[(.*?]*), (.*?]*)\]\]$"),
 }
 
-_MODULE_FILTERS: dict[str, re.Pattern] = {
-    # init matches only in the package root (__init__.py)
-    "init": re.compile(r"^homeassistant\.components\.\w+$"),
-    # any_platform matches any platform in the package root ({platform}.py)
-    "any_platform": re.compile(
-        f"^homeassistant\\.components\\.\\w+\\.({'|'.join([platform.value for platform in Platform])})$"
-    ),
-    # application_credentials matches only in the package root (application_credentials.py)
-    "application_credentials": re.compile(
-        r"^homeassistant\.components\.\w+\.(application_credentials)$"
-    ),
-    # backup matches only in the package root (backup.py)
-    "backup": re.compile(r"^homeassistant\.components\.\w+\.(backup)$"),
-    # cast matches only in the package root (cast.py)
-    "cast": re.compile(r"^homeassistant\.components\.\w+\.(cast)$"),
-    # config_flow matches only in the package root (config_flow.py)
-    "config_flow": re.compile(r"^homeassistant\.components\.\w+\.(config_flow)$"),
-    # device_action matches only in the package root (device_action.py)
-    "device_action": re.compile(r"^homeassistant\.components\.\w+\.(device_action)$"),
-    # device_condition matches only in the package root (device_condition.py)
-    "device_condition": re.compile(
-        r"^homeassistant\.components\.\w+\.(device_condition)$"
-    ),
-    # device_tracker matches only in the package root (device_tracker.py)
-    "device_tracker": re.compile(r"^homeassistant\.components\.\w+\.(device_tracker)$"),
-    # device_trigger matches only in the package root (device_trigger.py)
-    "device_trigger": re.compile(r"^homeassistant\.components\.\w+\.(device_trigger)$"),
-    # diagnostics matches only in the package root (diagnostics.py)
-    "diagnostics": re.compile(r"^homeassistant\.components\.\w+\.(diagnostics)$"),
+_MODULE_REGEX: re.Pattern[str] = re.compile(r"^homeassistant\.components\.\w+(\.\w+)?$")
+_PLATFORMS: list[str] = [platform.value for platform in Platform]
```

Comment:
If you use a set-comprehension, please move `_PLATFORMS`. It doesn't really make sense between two regex constants. Maybe above `TypeHintMatch`?

Reply:
done